crates/astria-sequencer/src/mempool.rs

```diff
@@ -20,7 +20,9 @@ pub(crate) struct TransactionPriority {
 
 impl TransactionPriority {
     fn nonce_diff(&self) -> u32 {
-        self.transaction_nonce - self.current_account_nonce
+        self.transaction_nonce
+            .checked_sub(self.current_account_nonce)
+            .expect("transaction nonce should not be less than current account nonce")
```

Comment:
```suggestion
            .expect("invariant violated: transaction nonce should not be less than current account nonce")
```

Reply:
Done in [44c99af](https://github.com/astriaorg/astria/pull/1081/commits/44c99affc52d66ee3134fa1bbc23f66d21bb632a).